### PR TITLE
[codesign + open to suggestions] log warning instead of exception when metadata isn't present

### PR DESCRIPTION
**Pull request by XilaiZhang**
open to suggestions, we have two options here:
1. enforce every artifact to have both `entitlements.txt` and `without_entitlements.txt` as metadata, and throw exception when any of the metadata is missing.
2. allow artifact to have a single or no metadata file, to imply that the artifact does not have any binary files to be codesigned (with/without entitlements). In this case we will convert exception to warning.

### CODE REVIEW

**Comment by XilaiZhang** [comment]
maybe @christopherfujino and @godofredoc would have some insights on which approach would be better. Thank you!

**Comment by godofredoc** [comment]
@XilaiZhang should we close or land this PR? is still relevant?

File: codesign/lib/src/file_codesign_visitor.dart [hunk 1]
```diff
@@ -317,8 +317,10 @@ update these file paths accordingly.
         ? fileSystem.path.join(parent.path, 'entitlements.txt')
         : fileSystem.path.join(parent.path, 'without_entitlements.txt');
     if (!(await fileSystem.file(entitlementFilePath).exists())) {
-      throw CodesignException('$entitlementFilePath not found \n'
-          'make sure you have provided them along with the engine artifacts \n');
+      log.warning('$entitlementFilePath not found. '
```
Comment (christopherfujino) [inline codesign/lib/src/file_codesign_visitor.dart:320]: The result is that no binaries will be signed, right? I think we should make that clear in this warning message (it was not clear to me until I read the PR description).

Comment (XilaiZhang) [reply codesign/lib/src/file_codesign_visitor.dart:320]: Good advice! Added a line informing user no binaries will be signed.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**